docs/integration/kafka.md

```diff
@@ -1,8 +1,191 @@
+```
+
+**You need to restart your Kafka consumer after the changes above.**
+
+## Collected Metrics
+
+The Sematext Kafka monitoring agent collects the following metrics.
+
+### Operating System
+
+- CPU usage
+- CPU load
+- Memory usage 
+- Swap usage
+- Disk space used
+- I/O Reads and Writes
+- Network traffic
+
+![](https://sematext.com/wp-content/uploads/2019/05/d_kafka_cpu.png)
+
+### Java Virtual Machine
+
+- Garbage collectors time and count
+- JVM pool size and utilization
+- Threads and daemon threads
+- Files opened by the JVM
+
+![](https://sematext.com/wp-content/uploads/2019/05/d_kafka_gc.png)
+
+### Kafka 
+
+- Partitions, leaders partitions, offline partitions, under replicated partitions
+- Static broker lag
+- Leader elections, unclean leader elections, leader elections time
+- Active controllers
+- ISR/Log flush 
+- Log cleaner buffer utilization, cleaner working time, cleaner recopy
+- Response and request queues
+- Replica maximum lag, replica minimum fetch, preferred replicas imbalances
+- Topic messages in, topic in/out, topic rejected, failed fetch and produce requests
+- Log segment, log size, log offset increasing 
+
+![](https://sematext.com/wp-content/uploads/2019/05/d_kafka_broker.png)
+
+### Kafka Producer
+
+- Batch size, max batch size
+- Compression rate 
+- Buffer available bytes 
+- Buffer pool wait ratio
+- I/O time, I/O ratio, I/O wait time, I/O wait ratio
+- Connection count, connection create rate, connection close rate, network I/O rate
+- Record queue time, send rate, retry rate, error rate, records per request, record size, responses rate, requests size and maximum size
+- Nodes bytes IN rate, node bytes OUT rate, request latency and max latency, request rate, responses rate, requests size and maximum size
+- Topic compression rate, bytes rate, records send rate, records retries rate, records errors rate
+
+![](https://sematext.com/wp-content/uploads/2019/05/d_kafka_producer.png)
+
+### Kafka Consumer
+
+- Consumer lag
+- Fetcher responses, bytes, responses bytes
+- I/O time, I/O ratio, I/O wait time, I/O wait ratio
+- Connection count, connection create rate, connection close rate, network I/O rate
+- Consumed rate, records per request, fetch latency, fetch rate, bytes consumed rate, average fetch size, throttle maximum time
+- Assigned partitions, heartbeat maximum response time, heartbeat rate, join time and maximum join time, sync time and maximum sync time, join rate, syncs rate
+- Nodes bytes IN rate, node bytes OUT rate, request latency and max latency, request rate, responses rate, requests size and maximum size
+
+![](https://sematext.com/wp-content/uploads/2019/05/d_kafka_consumer_lag.png)
+
+## Troubleshooting
+
+If you are having issues with Sematext Monitoring, i.e. not seeing SolrCloud metrics, you can create a diagnostics package on any affected machines where SolrCloud Monitoring Agent installed by running:
```

Comment:
We can be DRY...
Oh, and there is a mention of SolrCloud here.

Reply:
Yeah, my fault copy+paste issue :) Adjusted.